server/writer/gzip.go

```diff
@@ -58,7 +58,9 @@ func (g *Gzip) Header() http.Header {
 }
 
 func (g *Gzip) WriteHeader(statusCode int) {
-	g.rw.Header().Add(VaryHeader, AcceptEncodingHeader)
```

Comment:
Does this break (cache?) things if its stay here for small non compressed responses?